src/builtins/functions.rs

```diff
@@ -6,117 +6,120 @@ use serde_json::value::{from_value, to_value, Value};
 use errors::{Error, Result};
 
 /// The global function type definition
-pub type GlobalFn = Box<Fn(HashMap<String, Value>) -> Result<Value> + Sync + Send>;
-
-pub fn make_range_fn() -> GlobalFn {
-    Box::new(move |args| -> Result<Value> {
-        let start = match args.get("start") {
-            Some(val) => match from_value::<usize>(val.clone()) {
-                Ok(v) => v,
-                Err(_) => {
-                    return Err(Error::msg(format!(
-                    "Global function `range` received start={} but `start` can only be a number",
-                    val
-                )))
-                }
-            },
-            None => 0,
-        };
-        let step_by = match args.get("step_by") {
-            Some(val) => match from_value::<usize>(val.clone()) {
-                Ok(v) => v,
-                Err(_) => {
-                    return Err(Error::msg(format!(
-                    "Global function `range` received step_by={} but `step` can only be a number",
+pub trait GlobalFn {
```

Comment:
One of the other change is to completely remove the `global` part of global function from the API. since there are no local functions in Tera, it doesn't make too much sense.
Of course in this case we can't use `Fn` as it exists in the std...

Maybe `TeraFunction`, `TeraFilter` and `TeraTest`? Or `Test`, `Filter` and `Function` and recommend people to use them prefixed like `tera::Test` etc to avoid conflicts. I'm tempted by the second approach but not convinced at all

Reply:
Agreed